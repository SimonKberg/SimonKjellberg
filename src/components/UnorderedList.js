// @flow strict
import styled from 'react-emotion'
import { theme } from 'styled-tools'

export const List = styled('ul')`
  list-style-type: none;
  padding: 0;
  margin-top: 1em;
  margin-bottom: 1em;
  margin-left: 1.5ch;
`

export const ListItem = styled('li')`
  position: relative;

  &::before {
    content: '*';
    position: absolute;
    left: -1.5ch;
    color: ${theme('color.muted')};
  }

  & > ${List} {
    margin-top: 0;
    margin-bottom: 0;
  }
`
