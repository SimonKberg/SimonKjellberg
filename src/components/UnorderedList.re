open Utils;

let styles =
  Css.(
    style([
      listStyleType(`none),
      padding(zero),
      marginTop(em(1.0)),
      marginBottom(em(1.0)),
      marginLeft(ch(1.5)),
    ])
  );

[@react.component]
let make = (~className="", ~innerRef=?, ~children) => {
  <ul className={cn([styles, className])} ref=?innerRef> children </ul>;
};

let default = make;

module Jsx2 = {
  let component = ReasonReact.statelessComponent(__MODULE__);
  let make = (~className=?, ~innerRef=?, children) => {
    let children = React.array(children);
    ReasonReactCompat.wrapReactForReasonReact(
      make,
      makeProps(~className?, ~innerRef?, ~children, ()),
      children,
    );
  };
};
