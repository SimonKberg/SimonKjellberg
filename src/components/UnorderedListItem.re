open Utils;

let styles =
  Css.(
    style([
      position(`relative),
      before([
        contentRule("*"),
        position(`absolute),
        left(ch(-1.5)),
        color(Theme.Color.muted),
      ]),
      selector(
        "& > ." ++ UnorderedList.styles,
        [marginTop(zero), marginBottom(zero)],
      ),
    ])
  );

[@react.component]
let make =
  forwardDOMRef((~className=?, ~style=?, ~children, domRef) =>
    <li className={cn([styles, className |? ""])} ?style ref=?domRef>
      children
    </li>
  );

let default = make;
