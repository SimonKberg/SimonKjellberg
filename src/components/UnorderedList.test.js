import * as React from 'react'
import renderer from 'react-test-renderer'
import UnorderedList from './UnorderedList.bs'
import UnorderedListItem from './UnorderedListItem.bs'

test('UnorderedList', () => {
  const tree = renderer
    .create(
      <UnorderedList>
        <UnorderedListItem>Foo</UnorderedListItem>
        <UnorderedListItem>Bar</UnorderedListItem>
        <UnorderedListItem>
          <UnorderedList>
            Baz
            <UnorderedListItem>Qux</UnorderedListItem>
            <UnorderedListItem>Quux</UnorderedListItem>
          </UnorderedList>
        </UnorderedListItem>
      </UnorderedList>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
