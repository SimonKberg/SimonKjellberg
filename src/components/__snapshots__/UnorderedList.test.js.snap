// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnorderedList 1`] = `
.emotion-4 {
  list-style-type: none;
  padding: 0;
  margin-top: 1em;
  margin-bottom: 1em;
  margin-left: 1.5ch;
}

.emotion-0 {
  position: relative;
}

.emotion-0::before {
  content: "*";
  position: absolute;
  left: -1.5ch;
  color: #c0c0c0;
}

.emotion-0 > .emotion-4 {
  margin-top: 0;
  margin-bottom: 0;
}

<ul
  className="emotion-4"
>
  <li
    className="emotion-0"
  >
    Foo
  </li>
  <li
    className="emotion-0"
  >
    Bar
  </li>
  <li
    className="emotion-0"
  >
    <ul
      className="emotion-4"
    >
      Baz
      <li
        className="emotion-0"
      >
        Qux
      </li>
      <li
        className="emotion-0"
      >
        Quux
      </li>
    </ul>
  </li>
</ul>
`;
